app/controllers/shared/api.php

```diff
@@ -592,3 +593,19 @@
             throw new Exception(Exception::GENERAL_USAGE_DISABLED);
         }
     });
+
+App::init()
+    ->groups(['restrict'])
+    ->inject('request')
+    ->inject('geodb')
+    ->action(function (Request $request, Reader $geodb) {
+        if (!empty(app::getEnv('_APP_RESTRICTED_COUNTRIES', ''))) {
+            $countries = explode(',', App::getEnv('_APP_RESTRICTED_COUNTRIES', ''));
+            $record = $geodb->get($request->getIP());
+            $country = $record['country']['iso_code'];
+            $countryName = $record['country']['names']['en'];
+            if (in_array($country, $countries)) {
+                throw new Exception(Exception::GENERAL_ACCESS_FORBIDDEN, "Sorry, access from $countryName is restricted");
```

Comment:
We need to return a 503 service unavailable error in this case. Check the errors.php file to see if we have something like that already 👍

Reply:
We don't have any relevant 503 errors at the moment. We'd need a new one.

Reply:
Creating one with 451 as discussed